ru/turing/courses/lesson3/VersionA/Registry.java

```diff
@@ -0,0 +1,252 @@
+package ru.turing.courses.lesson3.VersionA;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+
+public class Registry {
+    public static class Animal {
+        private Map<String, Animal> storage = new HashMap<>();
+
+        /**
+         * add(name, animal)
+         * добавить в реестр новую запись
+         *
+         * @param name
+         * @param animal
+         */
+        public void add(String name, Animal animal) {
+            storage.put(name, animal);
+        }
+
+        /**
+         * removeByKey(String name)
+         * удалить из реестра запись по заданному ключу. Возвращать должен значение записи
+         *
+         * @param name
+         * @param <T>
+         * @return
+         */
+        public <T> T removeByKey(String name) {
+            T value = (T) storage.get(name);
+            storage.remove(name);
+            return value;
+        }
+
+        /**
+         * clear() - полностью очистить реестр
+         */
+        public void clear() {
+            storage.clear();
+        }
+
+        /**
+         * removeByValue(animal)
+         * удалить из реестра все записи по заданному значению
+         * Возвращать должен List ключей удаленных записей
+         *
+         * @param animal
+         * @return
+         */
+        public List<String> removeByValue(Animal animal) {
+            /**
+             * Список ключей удаленных записей по заданному значению
+             */
```

Comment:
```suggestion
             //Список ключей удаленных записей по заданному значению
```
А вот здесь, при однострочном пояснении внутренней логики метода, javadoc не нужен